packages/shared-ux/file/file_upload/impl/src/upload_state.ts

```diff
@@ -97,15 +97,22 @@ export class UploadState {
     return this.uploading$.getValue();
   }
 
-  private validateFiles(files: File[]): undefined | string {
-    if (
-      this.fileKind.maxSizeBytes != null &&
-      files.some((file) => file.size > this.fileKind.maxSizeBytes!)
-    ) {
-      return i18nTexts.fileTooLarge(String(this.fileKind.maxSizeBytes));
+  private readonly validateFile = (file: File): void => {
+    const fileKind = this.fileKind;
+
+    if (fileKind.maxSizeBytes != null && file.size > this.fileKind.maxSizeBytes!) {
+      const message = i18nTexts.fileTooLarge(String(this.fileKind.maxSizeBytes));
+      throw new Error(message);
     }
-    return;
-  }
+
+    if (fileKind.allowedMimeTypes != null && !fileKind.allowedMimeTypes.includes(file.type)) {
```

Comment:
Could there be an edge case with not a direct match? 
for example, could fileKind.allowedMimeTypes have something like  `image/*`, but file has `image/png` ?

Reply:
No, not currently at least, we don't support wildcards.